core/src/main/java/com/novoda/noplayer/exoplayer/ExoPlayerTwoImpl.java

```diff
@@ -30,14 +30,13 @@ public ExoPlayerTwoImpl(ExoPlayerTwoFacade facade) {
         heart = Heart.newInstance(onHeartbeat);
 
         prepareForwarder = new OnPrepareForwarder(getPreparedListeners(), this);
-        facade.addListener(prepareForwarder);
-        facade.addListener(new OnCompletionForwarder(getCompletionListeners()));
-        facade.addListener(new OnErrorForwarder(this, getErrorListeners()));
-        facade.addListener(new BufferStateForwarder(getBufferStateListeners()));
-        facade.addListener(new VideoSizeChangedForwarder(getVideoSizeChangedListeners()));
-
-        facade.setBitrateChangedListeners(getBitrateChangedListeners());
-        facade.setInfoListeners(getInfoListeners());
+        facade.addForwarder(prepareForwarder);
```

Comment:
for convenience you could create a `... addForwarder(Forwarder... forwarder)` method